src/engine/racer.rs

```diff
@@ -117,19 +115,19 @@ impl<'a> SemanticEngine for Racer<'a> {
             Err(poisoned) => poisoned.into_inner()
         };
         let session = Session::from_path(&cache, ctx.query_path(), ctx.query_path());
-        let (query_src, pos, path) = self.build_racer_args(ctx, &session);
+        let (file, pos, path) = self.build_racer_args(ctx, &session);
 
-        let matches = ::racer::core::complete_from_file(query_src, path, pos, &session);
+        let matches = ::racer::core::complete_from_file(&file.code, path, pos, &session);
 
-        let completions = matches.map(|m| {
+        let completions = matches.filter_map(|m| {
             let (line, col) = {
-                let found_src = &session.load_file(m.filepath.as_path());
-                let found_src_str = &found_src.src.code[..];
-                // TODO This can panic if the position is out of bounds :(
-                point_to_coords(found_src_str, m.point)
```

Comment:
There are a few things here that I'm not totally correct I got right. Why were we calling `load_file` again to get the source, isn't it the same as the old `query_src`?

It seems to work fine as is right now, instead of reloading the file, though maybe the match can be in other file or something?